### PR TITLE
mongodb: update to 3.6.2

### PR DESCRIPTION
**Pull request by emcrisostomo**
###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.13.2
Xcode 9

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by emcrisostomo** [comment]
@ryandesign, here's my first shot at updating `mongodb`. This should definitely be review by you. I have the following questions:

  * Upstream has documented conflicting build requirements (https://github.com/mongodb/mongo/blob/master/docs/building.md). On the one hand, at the beginning of the file they state that only `pyyaml` and `typing` python modules are required. Then, they say we should `pip2 install -r buildscripts/requirements.txt`. This file (https://github.com/mongodb/mongo/blob/master/buildscripts/requirements.txt) currently lists: 
 
```
jira == 1.0.10
pyyaml == 3.11
unittest-xml-reporting == 2.1.0
# Linters
yapf == 0.16.0
mypy == 0.501 ; python_version > "3"
# typing in Python 2 for mypy
typing == 3.6.1; python_version < "3"
pylint == 1.6.5
pydocstyle == 1.1.1
# resmoke.py
-r resmokelib/requirements.txt
# generate_error_codes.py
cheetah3 == 3.0.0; python_version < "3"
```

  * On the one hand I'm not sure how we can deal with this kind of requirements in a port (actually, we don't even have a port for all of them: I gave a shot at building using `py-cheetah` and it worked, but we don't have `py-cheetah3` yet.
  * On the other hand, mongo builds just fine with the build dependencies I put in this change set.

Let me know your thoughts and let's see how I can help you get this done.

**Comment by emcrisostomo** [comment]
@ryandesign I created a submission for `py-cheetah3` #1110

**Comment by emcrisostomo** [comment]
On a side note, can somebody help me understand what's going on the travis build? I've had a look at them but I'm not sure about what the problem is. I've already built this port on a bunch of machines of ours and it's working fine.

**Comment by mojca** [comment]
[The logs](say)
```
ImportError: No module named yaml
```

You probably forgot a dependency.

**Comment by emcrisostomo** [comment]
Thank you very much @mojca

File: databases/mongodb/Portfile [hunk 1]
```diff
@@ -25,19 +25,20 @@ homepage            https://www.mongodb.com/community
...
-                    size    39959937
+checksums           rmd160  f09df13045e05efd28ad85efe0217c24b2531295 \
+                    sha256  1573e6e172a6a1e559e1156b0f059f3639774a9f3db33790cb826c56e1398efd \
+                    size    39994543
 
-depends_build       port:scons
+depends_build       port:scons \
+                    port:py-typing \
+                    port:py-regex \
+                    port:py-cheetah
```
[the remark this answers is not in the document]
Comment (emcrisostomo) [reply databases/mongodb/Portfile]: Thanks @mf2k. I've updated the `Porfile` to use `py27-*` to stay in line with what upstream [suggests][building].

[building]: https://github.com/mongodb/mongo/blob/master/docs/building.md

**Comment by ryandesign** [comment]
I see the `--osx-version-min` SConscript flag is no longer supported, so it should just be removed. `-mmacosx-version-min=${macosx_deployment_target}` should not be added to `CCFLAGS` because in the absence of that the `MACOSX_DEPLOYMENT_TARGET` environment variable, which MacPorts base already sets, will get used. The only reason why the port was using the mongodb-specific `--osx-version-min` configure flag was that if we did not specify it, mongodb would choose a different value that we did not want it to choose.

In the error message about `--osx-version-min` no longer being supported, it says mongodb now requires OS X 10.10 or later. In a [bug report](https://jira.mongodb.org/browse/SERVER-31016?focusedCommentId=1671947&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-1671947), the developers stated the requirement was Xcode 8 or later (but Xcode 8 is not compatible with OS X 10.10). I haven't tested it on 10.9 or earlier or with Xcode 7 or earlier, but if it cannot be made to work by blacklisting older clang compilers or forcing newer SDKs, then the `pre-fetch` block will need to be adjusted. At that point the `minimum_xcodeversions` portgroup inclusion and invocation, and the `configure.sdk_version` stuff for 10.8, can be removed too. But it's worth testing on older systems before removing that.

**Comment by ryandesign** [comment]
How are the py27 dependencies used? Are we sure the build system knows it has to run the scripts that use those dependencies with `python2.7` and not with `/usr/bin/env python`? The build succeeds for me on Sierra (without the above py-cheetah3 port) and I do not have python27 selected, but I don't understand why it works.

**Comment by ryandesign** [comment]
The compiler blacklist should be changed to include `{clang < 802.0.42}` (the scons file specifically checks for this version) and also add `macports-clang-3.4` and `macports-clang-3.7`.

Using this, I can get past the compiler check on 10.9 and 10.10 but then it fails with:

```
EnvironmentError: No module named pkg_resources:
  File "/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_release_tarballs_ports_databases_mongodb/mongodb/work/mongodb-src-r3.6.0/SConstruct", line 3047:
    env.Tool('icecream')
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/scons-3.0.1/SCons/Environment.py", line 1788:
    tool = SCons.Tool.Tool(tool, toolpath, **kw)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/scons-3.0.1/SCons/Tool/__init__.py", line 118:
    module = self._tool_module()
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/scons-3.0.1/SCons/Tool/__init__.py", line 153:
    raise SCons.Errors.EnvironmentError(e)
```

I don't know what to do about that.

**Comment by ryandesign** [comment]
>     EnvironmentError: No module named pkg_resources:

This is fixed by adding a dependency on py27-setuptools. Successful builds so far on 10.9, 10.11, 10.12. Will test more versions later.

**Comment by emcrisostomo** [comment]
@ryandesign, thank you very much for all the testing you're doing. Are you taking care of modifying the Portfile or am I? I have no way to test the port with any of the older OS X/XCode/compiler configurations you mention in your previous comments.

**Comment by ryandesign** [comment]
I can make local changes to the port and commit them and close the pull request. If you want the pull request merged, I would ask you to make the changes to the pull request, because I don't understand git well enough to do so myself.

I still don't know what to do with the python dependencies, or understand how the build system knows which python to use.

**Comment by emcrisostomo** [comment]
Hi @ryandesign 

> I can make local changes to the port and commit them and close the pull request. If you want the pull request merged, I would ask you to make the changes to the pull request, because I don't understand git well enough to do so myself.

No problem whatsoever: you can go proceed.

> I still don't know what to do with the python dependencies, or understand how the build system knows which python to use.

Honestly, neither do I.

**Comment by mojca** [comment]
When in the correct branch you can do
```
git rebase -i HEAD~3
```
and replace `add` by `squash` in the last two commits to combine all the commits together.

**Comment by mojca** [comment]
I squashed the commits. Ryan, feel free to hit the button.

**Comment by emcrisostomo** [comment]
@mojca I guess that's why I suggested that @ryandesign do it: I'm not sure I would be able to reconstruct the exact changes needed from the comments. Furthermore, I couldn't test them because I haven't got any other macOS versions to try on

**Comment by emcrisostomo** [comment]
Hi @ryandesign, is this blocked on this issue?

> I still don't know what to do with the python dependencies, or understand how the build system knows which python to use.

**Comment by emcrisostomo** [comment]
I just bumped this PR to mongodb 3.6.2. Could you please check whether it can be merged or more work is required?

**Comment by pmetzger** [comment]
@emcrisostomo @ryandesign Is this now mergeable?

**Comment by ryandesign** [comment]
> Hi @ryandesign, is this blocked on this issue?

I'm satisfied that the the version of python that's being used is the one scons uses, so that's fine.

This is ready to be merged, if someone can please squash the two commits into one first. (I've tried, but I continue to be too stupid to understand how. I try to follow instructions, but run into errors, and Googling those errors leads me to suggestions that lead to further errors. If someone could rewrite [WorkingWithGit](https://trac.macports.org/wiki/WorkingWithGit) to assume absolutely no prior knowledge of git, with a step by step example starting at the very beginning, that might help.)

**Comment by emcrisostomo** [comment]
Thanks @ryandesign, I've just squashed the two commits into one.

**Comment by emcrisostomo** [comment]
@ryandesign @pmetzger We should also merge #1104 when we merge this one.

**Comment by pmetzger** [comment]
@emcrisostomo You should also squash #1104 and then I'll merge both.

**Comment by emcrisostomo** [comment]
@pmetzger Thanks. I've now squashed the commits in #1104 too.

**Comment by pmetzger** [comment]
Merged. Thank you so much for your patience, @emcrisostomo!

**Comment by emcrisostomo** [comment]
Thank you @pmetzger.

**Pull request merged**